docs/security/FreelistProtection.md

```diff
@@ -0,0 +1,107 @@
+# Protecting meta-data
+
+Corrupting an allocator's meta-data is a common pattern for increasing the power of a use-after-free or out-of-bounds write vulnerabilities.  If you can corrupt the allocator's meta-data, then you can take a control gadget in one part of a system, and use it to affect other parts of the system.  There are various approaches to protecting allocator meta-data, the most common are:
```

Comment:
Can you put each sentence on a separate line so that diffs are easy to read and it's easier to review please?